src/components/encoding-pane/encoding-shelf.tsx

```diff
@@ -40,14 +42,22 @@ interface EncodingShelfProps extends EncodingShelfPropsBase, EncodingShelfDropTa
 class EncodingShelfBase extends React.PureComponent<EncodingShelfProps, {}> {
 
   public render() {
-    const {id, connectDropTarget, fieldDef} = this.props;
+    const {id, connectDropTarget, fieldDef, handleAction} = this.props;
 
     const isWildcardShelf = isWildcard(id.channel);
     const channelName = isWildcardShelf ? 'any' : id.channel;
 
     return connectDropTarget(
       <div styleName={isWildcardShelf ? 'wildcard-shelf' : 'encoding-shelf'}>
-        <div styleName="shelf-label">{channelName}</div>
+        <div styleName="shelf-label">
+          <FieldCustomizer
```

Comment:
I think caret and tether related stuff should be a part of shelf. 

Basically, you can imagine in the future the EncodingShelf's caret open either a FieldCustomizer or ValueCustomizer depending on whether the channelDef is a fieldDef or not.   (Right now we of course only support fieldDef.)